### PR TITLE
Viem/migration

### PR DESCRIPTION
**Pull request by vampire-ab**

### CODE REVIEW

File: src/test/index.ts [hunk 1]
```diff
@@ -54,10 +54,10 @@ const paramRequest = {
 export async function TestGetQuoteRate() {
   let response;
   console.log('TestGetQuoteRate');
-  const { getQuoteRate: getQuoteRateUsingDZapClient } =
+  const { getQuoteRate: getQuoteRateUsingdzapClient } =
     useClient(TEST_CHAIN_ID);
   try {
-    response = await getQuoteRateUsingDZapClient(quoteRequests);
+    response = await getQuoteRateUsingdzapClient(quoteRequests);
```
Comment (dzaptech) [inline src/test/index.ts:60]: This should be DZap only not dzap - getQuoteRateUsingDZapClient

File: src/test/index.ts [hunk 2]
```diff
@@ -68,9 +68,9 @@ export async function TestGetQuoteRate() {
 export async function TestGetSwapParams() {
   let response;
   console.log('TestGetSwapParams');
-  const { getSwapParams: getParamUsingDZapClient } = useClient(TEST_CHAIN_ID);
+  const { getSwapParams: getParamUsingdzapClient } = useClient(TEST_CHAIN_ID);
   try {
-    response = await getParamUsingDZapClient(paramRequest);
+    response = await getParamUsingdzapClient(paramRequest);
```
Comment (dzaptech) [inline src/test/index.ts:73]: Same as above.

File: src/config/index.ts [hunk 3]
```diff
@@ -42,3 +34,11 @@ export const batchSwapIntegrators: {
     contract: '0x12480616436DD6D555f88B8d94bB5156E28825B1',
   },
 };
+
+export const ViemChains: { [key: number]: Chain } = {
```
Comment (dzaptech) [inline src/config/index.ts:38]: I think we should name it as Chains instead of ViemChains

File: package.json [hunk 4]
```diff
@@ -30,21 +30,23 @@
...
+    "mocha": "^10.2.0",
+    "nodemon": "^2.0.16",
     "prettier": "^3.1.0",
+    "ts-node": "^10.9.1",
     "typescript": "^5.3.2"
   },
   "dependencies": {
     "axios": "^1.2.1",
+    "bignumber.js": "^9.1.2",
     "ethers": "^5.7.2",
```
Comment (dzaptech) [inline package.json:48]: We need to remove ethers too.

File: src/hooks/useContract.ts [hunk 5]
```diff
@@ -1,21 +1,21 @@
-import { BigNumber, ethers, Signer } from 'ethers';
-import { SWAP_CONTRACTS } from 'src/config';
-import { SwapParamRequest } from 'src/types';
-import { Contract } from 'zksync-web3';
+import { SWAP_CONTRACTS, ViemChains } from 'src/config';
+import { HexString, SwapParamRequest } from 'src/types';
 import { fetchSwapParams } from '../api';
 import { getChecksumAddress, purgeSwapVersion } from '../utils';
+import { Client, WalletClient, getContract as fetchContract } from 'viem';
+import BigNumber from 'bignumber.js';
```
Comment (dzaptech) [inline src/hooks/useContract.ts:6]: Not related to this but we have a utils file for BigNumber, use it if you want to: https://github.com/DZapIO/DZapBackend/blob/develop/src/common/utils/big-number.ts

**Review by dzaptech — CHANGES_REQUESTED** [review]

File: src/test/index.ts (hunk 1 above)
Comment (vampire-ab) [reply src/test/index.ts:60]: There is no useClient file in hooks.

**Pull request merged**